bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/ghost/Ghost.java

```diff
@@ -0,0 +1,358 @@
+import org.jfree.util.Log;
+import scala.Option;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * Ghost specification class
+ */
+public final class Ghost extends Instrument implements BinningProvider, SpectroscopyInstrument {
+
+    private static final Logger Log = Logger.getLogger(Ghost.class.getName());
+
+    public static final String INSTR_DIR = "ghost";
+
+    public static final double PLATE_SCALE = 1.64;  // arcsec/mm
+
+    private static final double WellDepth = 350000;  // VENU has to confirm the correct value.
+    //private final TransmissionElement _blazeThrougthput;
+    //private final IFU_Trans _ifuTrans;
+    private final IFUComponent _ifu;
+    private final TransmissionElement _ghostResolution;
+
+    protected DetectorsTransmissionVisitor _dtv;
+
+    /**
+     * Related files will be in this subdir of lib
+     */
+    public static final String INSTR_PREFIX = "ghost_";
+
+    // Instrument reads its configuration from here.
+    private static final double AD_SATURATION = 65535;
+
+    protected Ghost[] _instruments;
+
+    protected final GhostParameters gp;
+    protected final ObservationDetails odp;
+
+    protected GhostGratingOptics _gratingOptics;
+    protected Detector _detector;
+    protected double _sampling;
+
+    /*
+      Provide value(s) of throughput for GHOST Cass unit and science cable, based on
+      results from Ross Zhelem 21 May 2019. The throughput includes Cass unit
+      (telecentricity lens and ADC) and fibre cable + microlens arrays at both ends,
+      but not any aperture losses
+     */
+    protected TransmissionElement _cableThrougthput;
+
+    protected TransmissionElement _fixedOptics;
+
+    protected TransmissionElement _resolutionElement;
+
+    private static final String[] DETECTOR_CCD_NAMES = {"BLUE", "RED"};
+
+    private static final String FILENAME = "ghost" + getSuffix();
+
+    // These are the limits of observable wavelength with this configuration.
+
+    private GhostType.DetectorManufacturer _ccd;
+
+
+    private GhostType.DetectorManufacturer _ccdColor;
+
+    private GhostSaturLimitRule _ghostSaturLimitWarning;  // GHOST-specific saturation limit warning
+
+    public Ghost(final GhostParameters gp, final ObservationDetails odp, final GhostType.DetectorManufacturer ccdColor) {
+        super(Site.GS, Bands.VISIBLE, INSTR_DIR, FILENAME);
+
+        Log.fine("Resolution: " + gp.resolution());
+        Log.fine("ccd_color: " + ccdColor.displayValue());
+        this.odp    = odp;
+        this.gp     = gp;
+
+        _ccdColor = ccdColor;
+        _detector = new Detector(getDirectory() + "/", getPrefix(),  _ccdColor.getManufacter(), _ccdColor.getManufacter());
+
+        _detector.setDetectorPixels(_ccdColor.getXsize());
+        addComponent(_detector);
+        _cableThrougthput = new TransmissionElement(getDirectory()+"/"+ Ghost.INSTR_PREFIX + "cable" +getSuffix());
```

Comment:
The file path could be constructed by the system, the slash there is system dependent